packages/core/src/interpreter.ts

```diff
@@ -193,7 +210,7 @@ export class Actor<
   // array of functions to defer
   private _deferred: Array<() => void> = [];
 
-  private update(state: InternalStateFrom<TLogic>): void {
+  private update(state: InternalStateFrom<TLogic>, event: TEvent): void {
```

Comment:
this would likely save you some casts:
```suggestion
  private update(state: InternalStateFrom<TLogic>, event: EventObject): void {
```

`TEvent` annotation isn't true here anyway since internal events don't belong to `TEvent`